controllers/neutronapi_controller.go

```diff
@@ -448,8 +449,21 @@ func (r *NeutronAPIReconciler) reconcileInit(
 			},
 		)
 		if err != nil {
+			instance.Status.Conditions.Set(condition.FalseCondition(
+				condition.TLSInputReadyCondition,
+				condition.ErrorReason,
+				condition.SeverityWarning,
+				condition.TLSInputErrorMessage,
+				err.Error()))
 			return ctrlResult, err
 		} else if (ctrlResult != ctrl.Result{}) {
+			// Marking the condition as Unknown because we are not returining
+			// an err, but comparing the ctrlResult: this represents an in
+			// progress operation rather than something that failed
+			instance.Status.Conditions.MarkUnknown(
+				condition.TLSInputReadyCondition,
+				condition.InitReason,
```

Comment:
Could we also use `RequestedReason` here and on L491 instead?  It seems to be what we use for these sorts of situations across the operators.  For example [1].

[1] https://github.com/openstack-k8s-operators/keystone-operator/blob/c47a8894eea5c6a5a7354e53f913a61032012835/controllers/keystoneapi_controller.go#L720-L722

Reply:
yeah, makes sense, updating in my next PS